### ISSUE TITLE
0.5.13 Multiple default exports crashing meteor and babel minifier

### ISSUE THREAD

**Issue by amsully** [davidhu2000/react-spinners#74]
**Describe the bug**

Error within UI after successful build:
![Screen Shot 2019-07-23 at 11 17 57 AM](https://user-images.githubusercontent.com/5599640/61724256-98381a00-ad3b-11e9-8920-552887bb7c30.png)
![Screen Shot 2019-07-23 at 11 17 46 AM](https://user-images.githubusercontent.com/5599640/61724268-9d956480-ad3b-11e9-8354-c5761f332c20.png)

Error thrown during babel minification process in meteor:
````Only one default export allowed per module.: 
 
export default Component; 
 
at maybeThrowMinifyErrorBySourceFile 
(packages/minifyStdJS/plugin/minify-js.js:96:26) 
at files.forEach.file (packages/minifyStdJS/plugin/minify-js.js:135:9) 
at Array.forEach (<anonymous>) 
at MeteorBabelMinifier.processFilesForBundle 
(packages/minifyStdJS/plugin/minify-js.js:118:9) 
 
 
While minifying app code: 
packages/minifyStdJS/plugin/minify-js.js:96:26: Babili minification error 
within packages/modules.js: 
node_modules/react-spinners/dist/BeatLoader.js 
 
Only one default export allowed per module.: 
 
export default Component; 
 
at maybeThrowMinifyErrorBySourceFile 
(packages/minifyStdJS/plugin/minify-js.js:96:26) 
at files.forEach.file (packages/minifyStdJS/plugin/minify-js.js:135:9) 
at Array.forEach (<anonymous>) 
at MeteorBabelMinifier.processFilesForBundle 
(packages/minifyStdJS/plugin/minify-js.js:118:9)
````

**To Reproduce**
1. Meteor app using version 1.8.1. Babel runtime 7.3.4.
2. Import react-spinner 0.5.13 into app and throws errors shown in screenshot.

**Temporary Solution**
Our solution was to downgrade to version 0.5.12 until this is resolved.

**Comment by davidhu2000** [comment]
hmm, interesting. 0.5.13 only have readme and changelog updates. It was a patch to fix the props table in readme. 

Just want to confirm, downgrading to 0.5.12 fixes your issue completely?

**Comment by alexmarmon** [comment]
@davidhu2000 I just came across this as well. Downgrading to 0.5.12 completely fixed for me.

**Comment by amsully** [comment]
Yes downgrading fixes the issue completely, our diff for the fix:

```
-    "react-spinners": "^0.5.4",
+    "react-spinners": "0.5.12",
```

`^0.5.4` was defaulting to `0.5.13` when building from scratch. Locally I have switched back and forth between `0.5.12` and `0.5.13` and replicated the problem. Only `0.5.13` has a problem.

I also get the same issue with `0.6.0-alpha.1`.

**Comment by openjck** [comment]
I'm seeing an apparently related issue when running jest tests with the latest version of *react-spinners* in our [ensemble](https://github.com/mozilla/ensemble) repo. Downgrading to 0.5.12 fixes it.

**Comment by 814k31** [comment]
0.5.13 breaks my tests as well, downgrading to 12 worked:
Jest output
```
<project>/node_modules/react-spinners/dist/index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import BarLoaderComponent from "./
BarLoader";
                                                                                                    ^^^^^^^^^^^^^^^^^^

    SyntaxError: Unexpected identifier

      1 | import React from 'react';
      2 | 
    > 3 | import { GridLoader } from 'react-spinners';
        | ^
      4 | 
      5 | import styles from './GridSpinner.module.scss';
```

**Comment by davidhu2000** [comment]
this is so weird. these are the only changes for `0.5.13`

https://github.com/davidhu2000/react-spinners/commit/53e943b40cff57a607e17268e986011c6cf4d93b

https://github.com/davidhu2000/react-spinners/commit/a51b9722e655e24c8e8ed7cec9828ae2edb4e88e

https://github.com/davidhu2000/react-spinners/commit/93ca51c9e57fe0a62aef64696f8fa9aa10592587

No idea how these commits are causing this issue.

Can you try to import like this and see if the issue persists? 

```
import GridLoader from 'react-spinners/GridLoader';
```

**Comment by 814k31** [comment]
@davidhu2000 importing like `import GridLoader from 'react-spinners/GridLoader';` seems to work for me

**Comment by davidhu2000** [comment]
sounds like there is an issue with the index.ts exports. I will look into that

**Comment by amsully** [comment]
Within a minimal replication repo, the change of import style resolved the problem, but our larger project has `react-spinners` throw an error when importing react.

**minimal replication repo/instructions:** https://github.com/amsully/react_spinner_issue

Error thrown when changing all imports in larger project: 
![Screen Shot 2019-07-26 at 9 57 29 AM](https://user-images.githubusercontent.com/5599640/61984948-bff5df00-afd4-11e9-9b93-2a8b509d652a.png)

**Comment by davidhu2000** [comment]
I figured it out. Looks like the `dist` folder accidentally got pushed up as part of this patch. That folder includes all the typescript stuff, and there look to be some bugs compiling the code. That explains why `v0.6.0-alpha.1` has similar problems. 

This is what caught my eye finally.

> [...]/ensemble/node_modules/react-spinners/dist/index.js:1

All the desired files should be in the root folder, to allow for individual imports. I cloned [ensemble](https://github.com/mozilla/ensemble), installed `v0.5.13`. Went into node modules and deleted `/dist` folder, then was able to run the tests successfully. 

I was in the middle of converting the project to typescript and must have switched between branches to push up this patch. Version `0.5.12` does not have this issue. 

This will be something I fix in `v0.6.x`. But for now, I don't think there are any actions to take. 

This seems to only affect meteor for some reason. I was able to minify without issues using react.

**Comment by davidhu2000** [comment]
looks like the problem is in tsconfig:

```
"module": "esnext",
```

need to be changed to 

```
"module": "commonjs",
```

**Comment by davidhu2000** [comment]
Going to keep this open until we can confirm this is resolved

**Comment by shaibam** [comment]
Will updating to the newest version fix it? or do I have to manually change the tsconfig?

**Comment by davidhu2000** [comment]
seems like there are a lot of users with this problem. so I'm going to deprecate this version. Hoping to release `0.6.0` soon, just promoted it to beta. 😄

**Comment by davidhu2000** [comment]
ok, `0.5.13` is deprecated. tagged `0.5.12` as latest.

**Comment by davidhu2000** [comment]
This has been fixed in `v0.6.0`
